opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -77,21 +77,21 @@ pub trait LogProcessor: Send + Sync + Debug {
 /// debugging and testing. For scenarios requiring higher
 /// performance/throughput, consider using [BatchLogProcessor].
 #[derive(Debug)]
-pub struct SimpleLogProcessor {
-    exporter: Mutex<Box<dyn LogExporter>>,
+pub struct SimpleLogProcessor<T: LogExporter> {
```

Comment:
Technically this is a breaking change, but I doubt it would impact anyone. It's probably okay to push it for `0.27.1`.